### PR TITLE
ci: add pipeline for storage network

### PR DESCRIPTION
**Pull request by yangchiu**
ci: add pipeline for storage network

For https://github.com/longhorn/longhorn/issues/4218

Signed-off-by: Yang Chiu <yang.chiu@suse.com>

### CODE REVIEW

**Comment by yangchiu** [comment]
@longhorn/qa Please help approve this PR to move things forward. Thank you!

**Comment by yangchiu** [comment]
@longhorn/qa Please help approve this PR to move thing forward. Thank you!

File: pipelines/storage_network/Jenkinsfile [hunk 1]
```diff
@@ -23,6 +25,7 @@ node {
 
     withCredentials([
         usernamePassword(credentialsId: CREDS_ID, passwordVariable: 'AWS_SECRET_KEY', usernameVariable: 'AWS_ACCESS_KEY'),
+        string(credentialsId: 'DO_CREDS', variable: 'DO_TOKEN'),
```
Comment (innobead) [inline pipelines/storage_network/Jenkinsfile:28]: We should not use digital ocean resources anymore. Let's remove them used in the current test code in another PR.

File: pipelines/storage_network/Dockerfile.setup [hunk 2]
```diff
@@ -25,10 +25,7 @@ RUN wget -q https://storage.googleapis.com/kubernetes-release/release/$KUBECTL_V
     wget -q "https://github.com/mikefarah/yq/releases/download/${YQ_VERSION}/yq_linux_amd64" && \
     mv yq_linux_amd64 /usr/local/bin/yq && \
     chmod +x /usr/local/bin/yq && \
-    apk add openssl openssh-client ca-certificates git rsync bash curl jq python3 py3-pip && \
-    ssh-keygen -t rsa -b 4096 -N "" -f ~/.ssh/id_rsa && \
-    curl -fsSL -o get_helm.sh https://raw.githubusercontent.com/helm/helm/main/scripts/get-helm-3 && \
-    chmod 700 get_helm.sh && \
-    ./get_helm.sh
+    apk add openssl openssh-client ca-certificates git rsync bash curl jq python3 py3-pip aws-cli && \
+    ssh-keygen -t rsa -b 4096 -N "" -f ~/.ssh/id_rsa
```
Comment (innobead) [inline pipelines/storage_network/Dockerfile.setup:29]: kubectl version looks pretty old. update the versions of these dependencies if needed later.

**Review by innobead — APPROVED** [review]
In general, LGTM.

File: pipelines/storage_network/Dockerfile.setup (hunk 2 above)
Comment (innobead) [reply pipelines/storage_network/Dockerfile.setup:29]: There is version information scattered throughout many files. We should think of how to centralize version control to reflect all changes in one place for all pipelines.

Comment (yangchiu) [reply pipelines/storage_network/Dockerfile.setup:29]: There is a ticket for this: https://github.com/longhorn/longhorn/issues/6485.

I'll work on it later.

Comment (innobead) [reply pipelines/storage_network/Dockerfile.setup:29]: Very good.

File: pipelines/storage_network/Jenkinsfile (hunk 1 above)
Comment (yangchiu) [reply pipelines/storage_network/Jenkinsfile:28]: Removed in https://github.com/longhorn/longhorn-tests/pull/1775.

**Pull request merged**